go-controller/pkg/mock/mock_lsp.go

```diff
@@ -0,0 +1,301 @@
+package mockovn
+
+import (
+	"fmt"
+
+	goovn "github.com/ebay/go-ovn"
+	util "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"k8s.io/klog"
+)
+
+const (
+	LogicalSwitchPortAddresses        string = "LSPAddressesField"
+	LogicalSwitchPortOptions          string = "LSPOptionsField"
+	LogicalSwitchPortDynamicAddresses string = "LSPDynamicAddressesField"
+	LogicalSwitchPortExternalId       string = "LSPExternalIdsField"
+	LogicalSwitchPortPortSecurity     string = "LSPPortSecurityField"
+	FakeUUID                                 = "8a86f6d8-7972-4253-b0bd-ddbef66e9303"
+)
+
+// Get logical switch port by name
+func (mock *MockOVNClient) LSPGet(lsp string) (*goovn.LogicalSwitchPort, error) {
+	var lspCache MockObjectCacheByName
+	var ok bool
+	if lspCache, ok = mock.cache[LogicalSwitchPortType]; !ok {
+		klog.V(5).Infof("cache doesn't have any object of type %s", LogicalSwitchPortType)
+		return nil, goovn.ErrorSchema
+	}
+	var port interface{}
+	if port, ok = lspCache[lsp]; !ok {
+		return nil, goovn.ErrorNotFound
+	}
+	if lspRet, ok := port.(*goovn.LogicalSwitchPort); ok {
+		return lspRet, nil
+	}
+	return nil, fmt.Errorf("invalid object type assertion for %s", LogicalSwitchPortType)
+}
+
+// Add logical port PORT on SWITCH
+func (mock *MockOVNClient) LSPAdd(ls string, lsp string) (util.OVNCommandInterface, error) {
```

Comment:
This won't take effect until the command is executed, right? So the functions that manipulate the cache should be in a closure.

Reply:
Yeah, it doesn't take effect till command is executed. It's really about building higher level constructs from the tool-box provided by the interface. The mock go-ovn interface framework (as is written now) doesn't really care about the command arguments passed to `Execute()`.

Reply:
done.